snapcraft/internal/pluginhandler.py

```diff
@@ -685,6 +671,24 @@ def _migrate_files(snap_files, snap_dirs, srcdir, dstdir, missing_ok=False,
 
         fixup_func(dst)
 
+def _organize_filesets(fileset, base_dir):
+    for key in fileset:
+        src = os.path.join(base_dir, key)
+        dst = os.path.join(base_dir, fileset[key])
+
+        if os.path.isfile(dst):
+            raise EnvironmentError(
+                'Stepping over existing file for organization %r',
+                os.path.relpath(dst, base_dir))
+
+        if os.path.isdir(src):
+            file_utils.link_or_copy_tree(src, dst)
+            # TODO create alternate organization location to avoid
+            # deletions.
```

Comment:
Yeah, and it'd be ideal if file-overwriting was caught by the conflict detector instead of us having to do it here.